### PR TITLE
Adding Pimte area detector devices 

### LINKED ISSUE #428: create CCD pimte area detector device
Pimte is one of the few CCD detector used around diamond and it is useful to be able to use them in bluesaky , it could also serve as the base class for p99 CCD detector #381 

## Acceptance Criteria
- [x]  Create an ophyd-async  device for pimte
- [x]  1.  The detector should be stageable 
             Able to change Acquistion setting, exposure time trigger mode, eg things that need to be change during each data collection                     
- [x]  2.  Should be able to change Diver setting, chip temperature readout time etc.
- [x]  3. Should be able to write data to somewhere.
- [x]  5. should be able to be use in bluesky plan.

### Tasks
- [x]  Test what functionality of the detector has using epic PV, eg does the roi work etc etc.
- [x]  Go through the device on i22 to draw up a proper task list. (see Ref #332) 
- [x] Create the device
- Add extra functionality that needed in drivers like [pilatus_driver.py](https://github.com/bluesky/ophyd-async/blob/main/src/ophyd_async/epics/areadetector/drivers/pilatus_driver.py) 
- A controller much like [pilatus_controller.py](https://github.com/bluesky/ophyd-async/blob/main/src/ophyd_async/epics/areadetector/controllers/pilatus_controller.py) to deal with how the data collection is done setting trigger, count time etc....
- [x] Run it in bluesky
- [x] Device unit test
- Controller and drivers test is required. [example](https://github.com/bluesky/ophyd-async/tree/main/tests/epics/areadetector)  
- [x] IRL test
- Should be able to produce images with different exposure time, change all the setting that are required for the image itself.

##Ref
[i10 version ](https://github.com/Relm-Arrowny/i10-bluesky/issues/12)

### PR DESCRIPTION
**Pull request by Relm-Arrowny**
Fixes #428 

### Instructions to reviewer on how to test:
1. unit test should cover almost every time
2. If IRL test is needed there are three jupyter notebook script [here](https://github.com/Relm-Arrowny/i10-bluesky/blob/12-create-ophy-device-for-pimte-area-detector/tests/stubs/pimteAdBluesky.ipynb) that should cover most of the functionality needed. 

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)

### CODE REVIEW

**Comment by DiamondJoseph** [comment]
I've enabled the CI for you @Relm-Arrowny - do you need adding to the DLS organisation?

File: src/dodal/devices/areadetector/pimteAD.py
Comment (DiamondJoseph) [inline src/dodal/devices/areadetector/pimteAD.py:35]:
```suggestion

                super().__init__(
            PimteController(self.drv),
            HDFWriter(
                self.hdf,
                directory_provider,
                lambda: self.name,
                ADBaseShapeProvider(self.drv),
```

If not using the Stats plugin don't enforce its use, we need a more composable way of enabling plugins and using their signals

File: src/dodal/devices/areadetector/epics/pimte_controller.py
Comment (DiamondJoseph) [inline src/dodal/devices/areadetector/epics/pimte_controller.py:31]: Is deadtime exposure + 0.1s ? or is it a flat 0.1s?

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]

File: tests/devices/unit_tests/test_pimte.py
Comment (Relm-Arrowny) [reply tests/devices/unit_tests/test_pimte.py:28]: I am not sure what to do, I stolen it from #332  may be I should move it to somewhere common on this PR? or wait until the #332 merge and use that instead?

File: tests/devices/unit_tests/test_pimte.py
Comment (Relm-Arrowny) [reply tests/devices/unit_tests/test_pimte.py]: Moved this into conftest.py

File: src/dodal/devices/areadetector/epics/pimte_controller.py
Comment (Relm-Arrowny) [reply src/dodal/devices/areadetector/epics/pimte_controller.py]: checked the manual and change the dead time to the hardware limit.

https://www.princetoninstruments.com/wp-content/uploads/2020/04/PI-MTE3-System-Manual-Issue-1-4411-0162.pdf

**Comment by Relm-Arrowny** [comment]
cleaning up, the detector is current under repair will reopen as needed.

**Pull request closed without merging**
